server/chat-plugins/announcements.ts

```diff
@@ -3,27 +3,26 @@
  * By Spandamn
  */
 
-import {MinorActivity} from './poll';
-
 export interface AnnouncementOptions {
-	announcementNumber?: number;
+	activityNumber?: number;
 	source: string;
 	timeoutMins?: number;
 	timerEnd?: number;
 }
 
 export interface AnnouncementData extends AnnouncementOptions {
-	readonly activityId: 'announcement';
+	readonly activityid: 'announcement';
 }
 
-export class Announcement extends MinorActivity {
-	readonly activityId: 'announcement';
-	announcementNumber: number;
+export class Announcement extends Rooms.MinorActivity {
+	readonly activityid = 'announcement' as ID;
+	name = 'Announcement';
+	onRename = undefined;
```

Comment:
Why not just make this optional and not put the undefined declare here?

Reply:
it _is_ optional, TS is just being buggy